rakelib/pkg.rake

```diff
@@ -75,9 +75,13 @@ namespace :pkg do
       mkdir_p ".cargo"
       sh "cargo vendor --versioned-dirs --locked #{vendor_dir} >> #{cargo_config_path} 2>/dev/null"
 
-      final_gemspec.files += dirglob("**/.cargo/**/*")
-      final_gemspec.files += dirglob("./#{vendor_dir}/**/*")
-      final_gemspec.files.reject! { |f| File.directory?(f) }
+      vendor_files = dirglob("./#{vendor_dir}/**/*").reject { |f| File.directory?(f) }      
+      # Ensure that all vendor files have the right read permissions,
+      # which are needed to build the gem. 
+      # The permissions that we want are `.rw-r--r--`
+      vendor_files.each { |f| File.chmod(0644, f) }
```

Comment:
Have you considered something like `FileUtils.chmod("a+r", f)`? That way, we'd preserve existing permissions and only append to them, if e.g. a file was executable.

Looks like all RubyGem cares about is that [it's `& 0444`](https://github.com/ruby/ruby/blob/d2b87456f0e0046630dea9dd7fae4ec6324e9e92/lib/rubygems/specification_policy.rb#L232) (readable by user, group, other).

I don't know if that's desirable, but it's closer to the current behaviour.

Reply:
Added that in https://github.com/bytecodealliance/wasmtime-rb/pull/166/commits/4381070618fafe2b0c66e18a1af6c9e31e0cace8, and works as expected!